lib/Frontend/CompilerInvocation.cpp

```diff
@@ -3552,6 +3576,9 @@ bool CompilerInvocation::parseArgs(
     return true;
   }
 
+  if (FrontendOpts.PrintTargetInfo)
+    setRuntimeResourcePath(computeRuntimeResourcePathForTargetInfo());
```

Comment:
This just fixes the narrow problem of passing the right path for the Swift core modules back to `swift-driver`, but I raised a larger concern in the issue that motivated this pull, swiftlang/swift-driver#1562, that the `swift-frontend` is not looking in a passed in `-sdk` for those Swift core modules.

Can this pull be broadened to fix that too, especially since @compnerd has proposed [looking in `-sdk` primarily for those Swift core modules](https://github.com/swiftlang/swift/blob/main/docs/CrossCompilationModel.md)?

Right now, the way it works is that `swift-driver` queries `swift-frontend` for what paths to use then uses those to set `-sdk` and `-resource-dir` flags to pass back to the `swift-frontend` for compiling new Swift files. During compilation, `swift-frontend` only checks `-resource-dir` and adjacent to the compiler for the Swift core modules like the stdlib.

I think we should change `swift-frontend` to look in `-sdk` and adjacent to the compiler instead and stop passing `-resource-dir` to the `swift-frontend` unless one was specified to `swift-driver`. I noted in the linked issue that the legacy C++ Driver worked that way, ie it did not pass `-resource-dir` to the `swift-frontend` unless one was passed to the Driver first.

This would be in keeping with Saleem's plan to de-emphasize `-resource-dir` to look for Swift core modules and look in `-sdk` instead.